newIDE/app/src/ProjectsStorage/DownloadFileStorageProvider/index.js

```diff
@@ -18,13 +18,18 @@ export default ({
     onSaveProjectAs: (project: gdProject, fileMetadata: ?FileMetadata) => {
       return new Promise(resolve => {
         setDialog(() => (
-          <DownloadSaveAsDialog
-            onDone={() => {
-              closeDialog();
-              resolve({ wasSaved: false, fileMetadata });
-            }}
-            project={project}
-          />
+          <UnsavedChangesContext.Consumer>
+            {unsavedChangesManagement => (
+              <DownloadSaveAsDialog
+                onDone={() => {
+                  unsavedChangesManagement.sealUnsavedChanges();
```

Comment:
We don't want the file storage to be responsible for, or even aware of the existence of UnsavedChanges. Let's remove this and make MainFrame responsible for this.

Otherwise, we risk never sealing unsaved changes for other provides.
It's a burden that must not be on them